### PR TITLE
fix(sidekick): purge directory index (#648)

### LINKED ISSUE #648: Sidekick: publish should purge */ with every /index*
### Expected Behaviour
A URL ending with `*/index` or `*/index.html` can also be cached under `*/`. Publish should purge both URLs to make sure all users are seeing the latest version.

### Actual Behaviour
Publish currently only purges the original URL.

@davidnuescheler

### PR DESCRIPTION
**Pull request by rofe**
Fix #648

### CODE REVIEW

File: tools/sidekick/app.js [hunk 1]
```diff
@@ -415,6 +415,14 @@
           sk.showModal(`Publishing ${path}`, true);
           const resp = await sendPurge(config, path);
           if (resp.ok) {
+            if (path.endsWith('/')) {
+              // directory, also purge index(.html)
+              await sendPurge(config, `${path}index`);
```
Comment (kptdobe) [inline tools/sidekick/app.js:420]: Can't we run those in parallel to speed up the process ?

**Review by kptdobe — APPROVED** [review]

File: tools/sidekick/app.js (hunk 1 above)
Comment (rofe) [reply tools/sidekick/app.js:420]: We could... the purge is pretty quick however, it's the re-fetch that can take time...

Comment (rofe) [reply tools/sidekick/app.js]: Feels much faster indeed!

**Review by kptdobe — APPROVED** [review]
Don't you have a test for it that you need to extend ?

**Comment by rofe** [comment]
Added test

**Review by kptdobe — APPROVED** [review]

**Pull request merged**